sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -3,209 +3,334 @@
-        :keyword selects: List of setting selectors to filter configuration settings
-        :type selects: list[~azure.appconfigurationprovider.SettingSelector]
-        :keyword trim_prefixes: List of prefixes to trim from configuration keys
-        :type trim_prefixes: list[str]
-        :keyword key_vault_options: Options for resolving Key Vault references
-        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
-        """
-        provider = AzureAppConfigurationProvider()
-
-        key_vault_options = kwargs.pop("key_vault_options", None)
-
-        provider.__buildprovider(connection_string, endpoint, credential, key_vault_options)
-
-        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
-
-        provider._trim_prefixes = sorted(kwargs.pop("trimmed_key_prefixes", []), key=len, reverse=True)
-
-        provider._dict = {}
+    if (endpoint or credential) and connection_string:
+        raise ValueError("Please pass either endpoint and credential, or a connection string.")
 
-        secret_clients = key_vault_options.secret_clients if key_vault_options else {}
 
-        for select in selects:
-            configurations = provider._client.list_configuration_settings(
-                key_filter=select.key_filter, label_filter=select.label_filter
-            )
-            for config in configurations:
+    provider = _buildprovider(connection_string, endpoint, credential, key_vault_options, **kwargs)
+    provider._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
 
-                trimmed_key = config.key
-                # Trim the key if it starts with one of the prefixes provided
-                for trim in provider._trim_prefixes:
-                    if config.key.startswith(trim):
-                        trimmed_key = config.key[len(trim) :]
-                        break
+    if key_vault_options is not None and len(key_vault_options.secret_clients) > 0:
+        for secret_client in key_vault_options.secret_clients:
+            provider._secret_clients[secret_client.vault_url] = secret_client
 
-                if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
-                    secret = provider.__resolve_keyvault_reference(config, key_vault_options, secret_clients)
-                    provider._dict[trimmed_key] = secret
-                elif provider.__is_json_content_type(config.content_type):
-                    try:
-                        j_object = json.loads(config.value)
-                        provider._dict[trimmed_key] = j_object
-                    except json.JSONDecodeError:
-                        # If the value is not a valid JSON, treat it like regular string value
-                        provider._dict[trimmed_key] = config.value
-                else:
+    for select in selects:
+        configurations = provider._client.list_configuration_settings(
```

Comment:
Is it by design to access private members?

Reply:
Yes, it was a side affect of the move out of the provider class and into the module level method.